Mathlib/RingTheory/Polynomial/SmallDegreeVieta.lean

```diff
@@ -100,4 +111,58 @@ lemma aroots_quadratic_eq_pair_iff_of_ne_zero' [CommRing T] [Field S] [Algebra T
     X ^ 2 + C ((algebraMap T S) b) * X + C ((algebraMap T S) c) by simp]
   exact roots_quadratic_eq_pair_iff_of_ne_zero' ha
 
+/-
+Polynomial versions of results in `Algebra.QuadraticDiscriminant`
+-/
+section QuadraticDiscriminant
+
+variable [Field R] {a b c : R}
+
+/-- Roots of a quadratic equation. -/
+theorem isRoot_quadratic_iff [NeZero (2 : R)] (ha : a ≠ 0) {s : R}
+    (h : discrim a b c = s * s) (x : R) :
+    IsRoot (C a * X ^ 2 + C b * X + C c) x ↔ x = (-b + s) / (2 * a) ∨ x = (-b - s) / (2 * a) := by
+  rw [← quadratic_eq_zero_iff ha h]; simp [pow_two]
+
+/-- Root of a quadratic when its discriminant equals zero -/
+theorem isRoot_quadratic_iff_of_discrim_eq_zero [NeZero (2 : R)] (ha : a ≠ 0)
+    (h : discrim a b c = 0) (x : R) :
+    IsRoot (C a * X ^ 2 + C b * X + C c) x ↔ x = -b / (2 * a) := by
+  rw [← quadratic_eq_zero_iff_of_discrim_eq_zero ha h]; simp [pow_two]
+
+/-- A quadratic has no root if its discriminant has no square root. -/
+theorem not_isRoot_of_discrim_ne_sq (h : ∀ s : R, discrim a b c ≠ s^2) (x : R) :
+    ¬ IsRoot (C a * X ^ 2 + C b * X + C c) x := by
```

Comment:
I think this is half of a much more useful result:
```lean
/-- Quantifier elimination for a quadratic equation in characteristic `≠ 2`. -/
lemma Polynomial.exists_isRoot_quadratic_iff [NeZero (2 : R)] (ha : a ≠ 0) :
    (∃ x, IsRoot (C a * X ^ 2 + C b * X + C c) x) ↔ ∃ s, discrim a b c = s ^ 2 := by
```
I admit this means that people with `(2 : R) = 0` or `a = 0` are no longer served by this lemma but I still think this is the right API.

Reply:
We could deduce `exists_isRoot_quadratic_iff` from `not_isRoot_of_discrim_ne_sq` and `isRoot_quadratic_iff`?